### ISSUE TITLE
[FEAT][alignment]: add minimal RL batch schema and synthetic GRPO fixtures for kernel validation

### ISSUE THREAD

**Issue by inaniloquentee** [Flink-ddd/RL-Engine#27]
## Summary

Add a generic RL-shaped batch schema and deterministic synthetic fixtures for validating Kernel-Align kernels without requiring a full RL trainer, reward model, vLLM worker, DeepSpeed worker, or Ray cluster.

`fused_logp` should be the first consumer, but the fixture must be generic enough for future sampling, attention, masked reduction, and RL loss kernels.

## Why This Issue Exists

Kernel-Align is an RL post-training infrastructure project, but kernels currently have only narrow random-tensor tests. That misses key RL workload properties:

- prompt tokens and completion tokens have different semantics;
- padding and ignored tokens must be masked;
- group size `G` affects memory pressure;
- completion lengths vary across samples;
- current, old, and reference policy quantities are compared;
- rewards and advantages feed loss kernels;
- future kernels may need sequence metadata, not just logits/token ids.

A generic synthetic RL batch fixture lets every kernel test speak the same workload language.

## Scope

Create a lightweight fixture module, for example:

- `rl_engine/testing/rl_batch.py`, preferred if benchmarks and tests will both use it; or
- `tests/fixtures/rl_batch.py`, if it should stay test-only at first.

The fixture should support tensors and metadata such as:

- `input_ids`: `[batch_or_group, total_seq_len]`
- `position_ids`: optional `[batch_or_group, total_seq_len]`
- `attention_mask`: `[batch_or_group, total_seq_len]`
- `prompt_mask`: `[batch_or_group, total_seq_len]`
- `completion_mask`: `[batch_or_group, total_seq_len]` or `[batch_or_group, completion_len]`
- `token_ids`: selected next-token ids for logprob kernels
- `rewards`: synthetic scalar or per-token rewards
- `advantages`: `[batch_or_group, completion_len]`
- `old_logps`: `[batch_or_group, completion_len]`
- `ref_logps`: `[batch_or_group, completion_len]`
- optional `temperature`
- optional `row_indices` / `valid_indices` for compact valid-token kernel paths

Configurable dimensions:

- batch size or group size `G`;
- number of prompts;
- samples per prompt;
- prompt length;
- completion length;
- vocab size;
- hidden size, optional for future LM-head or attention tests;
- number of heads/head dimension, optional for attention tests;
- dtype;
- device;
- valid-token mask density;
- deterministic seed.

## Non-Goals

- Do not implement a real reward model.
- Do not implement a full GRPO/PPO/DPO trainer.
- Do not integrate vLLM, DeepSpeed, Ray, TRL, OpenRLHF, or verl.
- Do not optimize CUDA kernels in this issue.

## Suggested Implementation

Add a dataclass such as:

```python
@dataclass
class SyntheticRLKernelBatch:
    input_ids: torch.Tensor
    attention_mask: torch.Tensor
    prompt_mask: torch.Tensor
    completion_mask: torch.Tensor
    token_ids: torch.Tensor
    rewards: torch.Tensor
    advantages: torch.Tensor
    old_logps: torch.Tensor
    ref_logps: torch.Tensor
    valid_indices: torch.Tensor | None = None
    metadata: dict[str, Any] = field(default_factory=dict)
```

Add a generator such as:

```python
batch = make_synthetic_rl_kernel_batch(
    num_prompts=4,
    samples_per_prompt=8,
    prompt_len=32,
    completion_len=128,
    vocab_size=4096,
    valid_density=0.75,
    dtype=torch.float16,
    device="cuda",
    seed=0,
)
```

Add helper views:

- dense completion view;
- flattened valid-token view;
- compact row indices for kernels that skip masked tokens;
- metadata summary for benchmark CSV rows.

## Acceptance Criteria

- Tests can create deterministic RL-shaped batches on CPU and CUDA.
- Fixture supports prompt/completion/padding masks.
- Fixture supports group-style generation with multiple samples per prompt.
- Fixture can produce valid-token row indices for future compact kernels.
- Fixture does not require external model weights, datasets, or services.
- At least one unit test verifies shapes, masks, deterministic generation, and valid-token indexing.
- `fused_logp` can consume the fixture as the first example, without making the fixture logprob-only.

## Dependencies

- None.

## Follow-Up Users

- RL kernel reference operator issue.
- RL-shaped kernel benchmark issue.
- Minimal RL loss-step kernel test issue.
- `fused_logp_kernel.cu` optimization PR.
- Future sampling, attention, and loss kernel PRs.

**Comment by inaniloquentee** [comment]
This is the PR: https://github.com/Flink-ddd/Kernel-Align/pull/35
